regulation-worker/internal/delete/api/api.go

```diff
@@ -113,32 +148,39 @@ func (api *APIManager) deleteWithRetry(ctx context.Context, job model.Job, desti
 	if err != nil {
 		return model.JobStatus{Status: model.JobStatusFailed, Error: err}
 	}
-
-	var jobResp []JobRespSchema
-	if err := json.Unmarshal(bodyBytes, &jobResp); err != nil {
-		return model.JobStatus{Status: model.JobStatusFailed, Error: err}
-	}
-	jobStatus := getJobStatus(resp.StatusCode, jobResp)
-	pkgLogger.Debugf("[%v] Job: %v, JobStatus: %v", destination.Name, job.ID, jobStatus)
-
-	oauthErrJob, oauthErrJobFound := getOAuthErrorJob(jobResp)
-
-	if oauthErrJobFound && isOAuthEnabled {
-		if oauthErrJob.AuthErrorCategory == oauth.AUTH_STATUS_INACTIVE {
-			return api.inactivateAuthStatus(&destination, job, oAuthDetail)
+	respStatusCode := resp.StatusCode
+	respBodyBytes := bodyBytes
+	// Post response work to be done for OAuthV2
+	if isOAuth && api.IsOAuthV2Enabled {
+		transportResponse := oauthv2.TransportResponse{} // initing to prevent panic
+		// We don't need to handle it, as we can receive a string response even before executing OAuth operations like Refresh Token or Auth Status Toggle.
+		// It's acceptable if the structure of bodyBytes doesn't match the oauthv2.TransportResponse struct.
+		err = json.Unmarshal(bodyBytes, &transportResponse)
+		if err == nil && transportResponse.OriginalResponse != "" {
```

Comment:
Lets handle the err != nil case here first and go on to handle other cases?

Reply:
The comment is the reason why we're not handling err != nil case. Let me know if you need more clarification